src/reset.cpp

```diff
@@ -199,22 +199,29 @@ ResetPtr Reset::clone() const
     return r;
 }
 
-bool Reset::doEqual(const EntityPtr &other) const
+bool Reset::doEquals(const EntityPtr &other) const
 {
-    if (Entity::doEqual(other)) {
+    if (Entity::doEquals(other)) {
         auto reset = std::dynamic_pointer_cast<Reset>(other);
-        if (reset != nullptr && mPimpl->mOrder == reset->order() && compareMath(mPimpl->mResetValue, reset->resetValue()) && mPimpl->mResetValueId == reset->resetValueId() && compareMath(mPimpl->mTestValue, reset->testValue()) && mPimpl->mTestValueId == reset->testValueId()) {
-            bool equal = true;
-            if (mPimpl->mTestVariable != nullptr && !mPimpl->mTestVariable->equals(reset->testVariable())) {
-                equal = false;
-            } else if (mPimpl->mTestVariable == nullptr && reset->testVariable() != nullptr) {
-                equal = false;
-            } else if (mPimpl->mVariable != nullptr && !mPimpl->mVariable->equals(reset->variable())) {
-                equal = false;
-            } else if (mPimpl->mVariable == nullptr && reset->variable() != nullptr) {
-                equal = false;
+        if ((reset != nullptr) && mPimpl->mOrder == reset->order()
+                && areEqual(mPimpl->mResetValue, reset->resetValue())
+                && mPimpl->mResetValueId == reset->resetValueId()
+                && areEqual(mPimpl->mTestValue, reset->testValue())
+                && mPimpl->mTestValueId == reset->testValueId()) {
+            if (mPimpl->mTestVariable != nullptr
+                    && !mPimpl->mTestVariable->equals(reset->testVariable())) {
+                return false;
+            } else if ((mPimpl->mTestVariable == nullptr)
+                       && reset->testVariable() != nullptr) {
+                return false;
+            } else if ((mPimpl->mVariable != nullptr)
+                       && !mPimpl->mVariable->equals(reset->variable())) {
+                return false;
+            } else if ((mPimpl->mVariable == nullptr)
+                       && reset->variable() != nullptr) {
+                return false;
             }
```

Comment:
I am pretty sure this will fail our CI tests. When there is a `return` statement within an `if` statement then we cannot have an `else` clause. IOW, it should be (also added some "missing" parentheses here and there):
```c++
if ((mPimpl->mTestVariable != nullptr)
    && !mPimpl->mTestVariable->equals(reset->testVariable())) {
    return false;
}

if ((mPimpl->mTestVariable == nullptr)
    && (reset->testVariable() != nullptr)) {
    return false;
}

if ((mPimpl->mVariable != nullptr)
    && !mPimpl->mVariable->equals(reset->variable())) {
    return false;
}

if ((mPimpl->mVariable == nullptr)
    && (reset->variable() != nullptr)) {
    return false;
}
```

Reply:
This is not what is in the file, this is outdated.

Reply:
Yes, just came across your commit that does the above.